src/wui/portdockwaresdisplay.cc

```diff
@@ -70,14 +71,147 @@ AbstractWaresDisplay* create_portdock_wares_display(UI::Panel* parent,
+static char const* kPicWareMissing = "images/wui/buildings/low_priority_button.png";
+static char const* kNoWare = "images/wui/editor/no_ware.png";
+static const auto kEmptySlot = std::make_pair(Widelands::wwWARE, Widelands::INVALID_INDEX);
+
+struct PortDockAdditionalItemsDisplay : UI::Box {
+public:
+	PortDockAdditionalItemsDisplay(
+	   Widelands::Game& g, Panel* parent, bool can_act, PortDock& pd, const uint32_t capacity)
+	   : UI::Box(parent, 0, 0, UI::Box::Horizontal), game_(g), portdock_(pd), capacity_(capacity) {
+		assert(capacity_ > 0);
+		assert(portdock_.expedition_bootstrap());
+		assert(portdock_.expedition_bootstrap()->count_additional_queues() <= capacity_);
+		for (uint32_t c = 0; c < capacity_; ++c) {
+			UI::Box* box = new UI::Box(this, 0, 0, UI::Box::Vertical);
+
+			UI::Dropdown<std::pair<Widelands::WareWorker, Widelands::DescriptionIndex>>& d =
+			   *new UI::Dropdown<std::pair<Widelands::WareWorker, Widelands::DescriptionIndex>>(
+			      box, (boost::format("additional_%u") % c).str(), 0, 0, kWareMenuPicWidth, 8,
+			      kWareMenuPicHeight, _("Additional item"), UI::DropdownType::kPictorial,
+			      UI::PanelStyle::kWui, UI::ButtonStyle::kWuiSecondary);
+			d.add(_("(Empty)"), kEmptySlot, g_gr->images().get(kNoWare), true, _("(Empty)"));
+			for (Widelands::DescriptionIndex i : pd.owner().tribe().wares()) {
+				const Widelands::WareDescr& w = *pd.owner().tribe().get_ware_descr(i);
+				d.add(
+				   w.descname(), std::make_pair(Widelands::wwWARE, i), w.icon(), false, w.descname());
+			}
+			for (Widelands::DescriptionIndex i : pd.owner().tribe().workers()) {
+				const Widelands::WorkerDescr& w = *pd.owner().tribe().get_worker_descr(i);
+				d.add(
+				   w.descname(), std::make_pair(Widelands::wwWORKER, i), w.icon(), false, w.descname());
+			}
+			d.set_enabled(can_act);
+			d.selected.connect(boost::bind(&PortDockAdditionalItemsDisplay::select, this, c));
+
+			UI::Icon* icon = new UI::Icon(box, g_gr->images().get(kNoWare));
+			boxes_.push_back(box);
+			icons_.push_back(icon);
+			dropdowns_.push_back(&d);
+			box->add(&d);
+			box->add(icon);
+			add(box);
+		}
+		update_selection();
+	}
+
+	void think() override {
+		UI::Box::think();
+		if (!portdock_.expedition_bootstrap()) {
+			return die();
+		}
+		update_selection();
+
+		for (uint32_t c = 0; c < capacity_; ++c) {
+			const InputQueue* iq = portdock_.expedition_bootstrap()->inputqueue(c);
+			assert(!iq || (iq->get_max_size() == 1 && iq->get_max_fill() == 1));
+			icons_[c]->set_icon(g_gr->images().get(
+			   iq ?
+			      iq->get_filled() ? kPicWarePresent : iq->get_missing() ? kPicWareMissing :
+			                                                               kPicWareComing :
+			      kNoWare));
+			icons_[c]->set_tooltip(
+			   iq ?
+			      iq->get_filled() ?
+			      /** TRANSLATORS: Tooltip for a ware that is present in the building */
+			         _("Present") :
```

Comment:
Where are these strings in the UI? I can't find them.

Reply:
Those are the tooltips for the red/yellow/green icons below the additonal-item-dropdowns